src/FixedDataTableCellGroup.js

```diff
@@ -119,66 +120,109 @@ class FixedDataTableCellGroupImpl extends React.Component {
   constructor(props) {
     super(props);
     this._initialRender = true;
+    this._staticCells=[];
   }
 
   componentDidMount() {
     this._initialRender = false;
   }
+  shouldComponentUpdate() /*boolean*/ {
+    // Don't add PureRenderMixin to this component please.
+    return true;
+  }
 
   /**
    * Returns Object consisting of keys and widths of the columns in the current cell group.
    * @returns {{keys: [], widths: []}}
    */
   getCellGroupWidth = () => {
+  
     const { columns } = this.props;
     const cellGroupColumnWidths = {
       keys: [],
       widths: [],
     };
+    var colsToRender=this.props.colsToRender || [];
+   
     if (this.props.isHeader) {
-      for (let i = 0, j = columns.length; i < j; i++) {
-        const key = columns[i].props.columnKey || 'cell_' + i;
+      for (let i = 0, j = colsToRender.length; i < j; i++) {
+        var idx=colsToRender[i];
+        
+        if (idx === undefined) {
+          idx =
+            this._staticCells[i] && this._staticCells[i].props.index;
+    
+        }
+        
+        const key = columns[idx].props.columnKey || 'cell_' + idx;
         cellGroupColumnWidths.keys.push(key);
-        cellGroupColumnWidths.widths.push(columns[i].props.width);
+        cellGroupColumnWidths.widths.push(columns[idx].props.width);
       }
     }
     return cellGroupColumnWidths;
   };
+  componentWillUnmount() {
+    this._staticCells.length = 0;
+  }
+
 
   render() /*object*/ {
     var props = this.props;
     var columns = props.columns;
-    var cells = new Array(columns.length);
-    var contentWidth = sumPropWidths(columns);
-
+    var colsToRender=props.colsToRender || [];
+    
+    var isScrolling=props.isScrolling;
+   
+    var contentWidth = columns.length && (columns[columns.length-1].offset+ columns[columns.length-1].props.width);
```

Comment:
TODO calculate content width using interval tree , need to make 3 trees each for fixed,fixed right ,scrollable columns